plasmapy/diagnostics/tests/test_langmuir.py

```diff
@@ -1,40 +1,86 @@
-"""Tests for functions that use Langmuir analysis functions."""
+# coding=utf-8
+"""Tests for Langmuir probe analysis functions."""
 
 import numpy as np
+from astropy import units as u
+import pytest
+import plasmapy.constants as const
 
-from plasmapy.diagnostics.langmuir import (swept_probe_analysis,
-                                           obtain_EEDF,
-                                           obtain_EEPF)
-
-
-class Test_swept_probe_analysis():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Performing a swept probe analysis should return an error "
-                  "when zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEDF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEDF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEPF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEPF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
+from plasmapy.diagnostics.langmuir import (Characteristic,
+                                           swept_probe_analysis)
+
+N = 30  # length of test characteristic
+
+I_arr = np.random.rand(N) * u.A
+U_arr = np.random.rand(N) * u.V
+class Test__characteristic_errors:
+    r"""Test the Characteristic class constructor in langmuir.py"""
+    N = 30  # length of test characteristic
+    U_infarr = np.append(np.random.rand(N - 1), np.inf) * u.V
+    I_infarr = np.append(np.random.rand(N - 1), np.inf) * u.A
```

Comment:
Is this supposed to be an array of `inf`s or adding `inf`s to an array for testing how they're handled?